File: java/src/main/java/it/giannibombelli/refactor_challenge/Application.java
package it.giannibombelli.refactor_challenge;

public class Application {
    public static void main(String[] args) {
        final FizzBuzz fizzBuzz = new FizzBuzz();
        for (int number = 1; number <= 100; number++) {
            final String say = fizzBuzz.say(number);
            System.out.println(say);
        }
    }
}

File: kotlin/src/main/kotlin/it/giannibombelli/refactor_challenge/Application.kt
package it.giannibombelli.refactor_challenge

private val divisorAndWords = listOf(
    DivisorAndWord(3, "Fizz"),
    DivisorAndWord(5, "Buzz")
)

fun main(args: Array<String>) {
    val fizzBuzz = FizzBuzz(divisorAndWords)
    for (number in 1..100) {
        val say = fizzBuzz.say(number)
        println(say)
    }
}
